lib/content-negotiation.js

```diff
@@ -39,43 +48,55 @@ const jsonMarshall = (object) => {
 
 const mediaTypeMarshallers = {
     "application/octet-stream": {
-        unmarshall: (buffer) => buffer,
-        marshall: (any) => Buffer.from(any),
+        unmarshal: (buffer) => buffer,
+        marshal: (any) => Buffer.from(any),
     },
     "text/plain": {
-        unmarshall: (buffer) => textDecoder.decode(buffer),
-        marshall: (string) => textEncoder.encode("" + string),
+        unmarshal: (buffer, charset) =>
+            resolveDecoder(charset, () => "utf8").decode(buffer),
+        marshal: (string) => textEncoder.encode("" + string),
     },
     "application/json": {
-        unmarshall: jsonUnmarshall,
-        marshall: jsonMarshall,
+        unmarshal: jsonUnmarshall,
+        marshal: jsonMarshall,
     },
     "application/x-www-form-urlencoded": {
-        unmarshall: (buffer) => querystring.parse(textDecoder.decode(buffer)),
-        marshall: (object) => textEncoder.encode(querystring.stringify(object)),
+        unmarshal: (buffer) =>
+            querystring.parse(utf8TextDecoder.decode(buffer)),
+        marshal: (object) => textEncoder.encode(querystring.stringify(object)),
     },
     "application/cloudevents+json": {
-        unmarshall: jsonUnmarshall,
-        marshall: jsonMarshall,
+        unmarshal: jsonUnmarshall,
+        marshal: jsonMarshall,
     },
 };
 
-function canMarshall(type) {
+function resolveDecoder(initialCharset, defaultCharsetSupplier) {
+    const charset = initialCharset || defaultCharsetSupplier();
+    return new TextDecoder(charset);
+}
+
```

Comment:
dead code (if other suggestions are accepted)
```suggestion
```

Reply:
the initial idea was to allow to throw if a charset was not specified
let's forget this idea for now and go with the approach you suggested